File: LeetCode Solutions/Problems/696. Count Binary Substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int current = 1, previous = 0, result = 0;
        for(int i=1; i<s.length(); i++){
            if(s[i] == s[i-1]){
                current++;
            } else{
                result += min(current, previous), previous = current, current = 1;
            }
        }
        return result + min(current, previous);
    }
};

File: LeetCode Solutions/Problems/696. Count Binary Substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        current, previous, result = 1, 0, 0
        for i in range(1, len(s)):
            if s[i] == s[i-1]:
                current += 1
            else:
                result += min(current, previous)
                previous, current = current, 1
        return result + min(current, previous)